### PR TITLE
sway-fmt-v2 formatting should use tokens directly from sway-parse

### LINKED ISSUE #2072: `sway-fmt-v2` formatting should use `tokens` directly from `sway-parse`
Instead of something like: 

```Rust
formatted_code.push_str("}\n");
```

we should do:
```Rust
use sway_parse::token::Delimiter;

formatted_code.push_str(&format!("{}\n", Delimiter::Brace.as_close_char());
```

As seen in #2061

### PR DESCRIPTION
**Pull request by kayagokalp**
It would be better to wait for structs to get merged and fix that place as well before merging this. So until #2058 is merged I am leaving this as a draft. 

closes #2072.

### Todo
- [x] Merge and address #2058

### CODE REVIEW

**Comment by kayagokalp** [comment]
This should be good to go for a review.

**Review by eureka-cpu — DISMISSED** [review]
lgtm

File: sway-fmt-v2/src/items/item_enum.rs [hunk 1]
```diff
@@ -65,7 +65,9 @@ impl Format for ItemEnum {
                 // TODO: Improve handling this
                 formatted_code.push_str(&(0..required_alignment).map(|_| ' ').collect::<String>());
             }
-            formatted_code.push_str(" : ");
+            formatted_code.push(' ');
+            formatted_code.push(PunctKind::Colon.as_char());
+            formatted_code.push(' ');
```
Comment (mitchmindtree) [inline sway-fmt-v2/src/items/item_enum.rs:70]: Could also be written as:

```rust
write!(formatted_code, " {} ", PunctKind::Colon.as_char())?;
```

File: sway-fmt-v2/src/items/item_struct.rs [hunk 2]
```diff
@@ -99,7 +102,8 @@ fn format_struct(
         // Push a single whitespace before `{`
         formatted_code.push(' ');
         // Push open brace
-        formatted_code.push('{');
+        let open_brace = Delimiter::Brace.as_open_char();
+        formatted_code.push(open_brace);
         // Push a single whitespace after `{`
         formatted_code.push(' ');
```
Comment (mitchmindtree) [inline sway-fmt-v2/src/items/item_struct.rs:108]: Similarly here ^

File: sway-fmt-v2/src/utils/program_type.rs [hunk 3]
```diff
@@ -24,7 +24,6 @@ pub(crate) fn insert_program_type(push_to: &mut String, module_kind: ModuleKind)
             push_to.push_str(name.as_str());
         }
     };
-    push_to.push(';');
-    push_to.push('\n');
-    push_to.push('\n');
+    push_to.push(PunctKind::Semicolon.as_char());
+    push_to.push_str("\n\n");
```
Comment (mitchmindtree) [inline sway-fmt-v2/src/utils/program_type.rs:28]: Also here ^

**Review by mitchmindtree — APPROVED** [review]
Lg!

Left a couple of tips, but not necessary or particularly relevant to this PR

**Review by mitchmindtree — APPROVED** [review]

**Pull request merged**